cloudFiles/ingress-resoure.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-resource
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+    nginx.ingress.kubernetes.io/enable-cors: "true"
+    nginx.ingress.kubernetes.io/cors-allow-origin: <frontend-ip-address>
+spec:
+  rules:
+    - http:
+        paths:
+          - path: /api/questions
+            pathType: Prefix
+            backend:
+              service:
+                name: question-service
+                port:
+                  number: 4567
+          - path: /api/refresh
+            pathType: Prefix
+            backend:
+              service:
+                name: user-service
+                port:
+                  number: 4000
+          - path: /api/login
+            pathType: Prefix
+            backend:
+              service:
+                name: user-service
+                port:
+                  number: 4000
+          - path: /api/user
```

Comment:
I'm afraid this path will never let `/api/users` run because you match by prefix. Try this instead:

```
path: /api/user
pathType: Exact
backend:
  service:
    name: user-service
    port:
      number: 4000
```
More about this can be read [here](https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/).

Reply:
no worries, i have specified below /api/users and I believe prefix mean /api/user/*. So its anything after the /

Reply:
And the important part is they are under the same service, ingress only specify which service i use, as to the actual functionality, it still determined by the service itself